internal/cmd/iam/command_provider_delete.go

```diff
@@ -34,20 +30,46 @@ func (c *identityProviderCommand) newDeleteCommand() *cobra.Command {
 }
 
 func (c *identityProviderCommand) delete(cmd *cobra.Command, args []string) error {
-	provider, err := c.V2Client.GetIdentityProvider(args[0])
-	if err != nil {
+	if err := c.confirmDeletion(cmd, args); err != nil {
 		return err
 	}
 
-	promptMsg := fmt.Sprintf(errors.DeleteResourceConfirmMsg, resource.IdentityProvider, args[0], provider.GetDisplayName())
-	if _, err := form.ConfirmDeletion(cmd, promptMsg, provider.GetDisplayName()); err != nil {
+	deleteFunc := func(id string) error {
+		if err := c.V2Client.DeleteIdentityProvider(id); err != nil {
+			return err
+		}
+		return nil
```

Comment:
```suggestion
		return c.V2Client.DeleteIdentityProvider(id)
```

Reply:
I have to say, this is getting progressively more embarrassing with each one 😭